discussion-questions.md

```diff
@@ -1,8 +1,12 @@
 #  How much time did you spend on this assignment?
 
-_replace this text with your answer_
-
+I spent a total of approximately 5.5 hours on this assignment:
+- 2.5 hours on parts 1-4 of the assignment, installing and familiarizing myself with Scala
+- 2 hours on parts 5-6 of the assignment in building my own functions, adding fluency to the picture library, and modifying the tests accordingly
+- 1 hour on attempting to implement new functions, i.e. blur, bisectHorizontally, and bisectVertically. I did not have much trouble creating the bisecting functions, however, I spent a considerable amount of time trying to understand the algorithm for and attempting to comprehend which functions and methods to use to implement the Solarize and Blur functions.
 # Discussion questions / topics
 Below, add one or two things that you might want to discuss when we meet again.
 
-_replace this text with your discussion questions / topics_
\ No newline at end of file
+- In using Scala, I found many similarities between it and Python as well as Java. A question I wanted to ask was that why is it then that we use Scala, and not Python or Java, in implementing DSLs? Does it have some features which make it better-suited for this purpose?
```

Comment:
I hope that the most recent group session started to show why we're using Scala. Scala gives us more options for "removing the host-language flavor" than other general-purpose programming languages. We'll see some more of these options in the next group session.